### PR TITLE
Activate the fonts coming from the backend and not the data from the frontend

### LINKED ISSUE #60040: Fonts with spaces are not correctly applied
### Description

If I upload a font (woff2) that has one or more spaces in the name, this font is not declared correctly in the CSS in the frontend and is therefore not displayed.

Here is the concrete problem:

When I upload the font "Heineken Serif 18" in the new font library (The fonts I use are attached) the following CSS is created in the frontend:

`body {
--wp--preset--font-family--heineken-serif-18: Heineken Serif 18;
}`

Therefore, the font is not displayed: But it should be the following:

`body {
--wp--preset--font-family--heineken-serif-18: "Heineken Serif 18";
}`

The strange thing is that I installed a second font directly from Google Fonts as a test (see attached screenshot). The font is called "PT Sans", and this is correctly declared with quotation marks.

### Step-by-step reproduction instructions

1. Go to Theme Customizer
2. Open The Font Library
3. Upload the Fonts
4. Set the Font as Default

### Screenshots, screen recording, code snippet

Bildschirmfoto vom 2024-03-20 14-57-13

### Environment info

WordPress 6.5-RC3-57856,
Ubuntu 22.04 Desktop System in Firefox and Google Chrome

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by matiasbenedetto**
## What?
Activate the fonts coming from the backend and not the data from the frontend

## Why?
In some cases, as in https://github.com/WordPress/gutenberg/issues/60040, the data coming from the frontend can have non-properly formatted the `font-family` property. In the case of the issue linked `Heineken Serif 18` instead of `"Heineken Serif 18"`.

Fixes: https://github.com/WordPress/gutenberg/issues/60040

## How?
In this PR we use the font family and font face data coming from the API to activate the fonts (add to the Global Styles).

## Testing Instructions
1. Follow the instructions from https://github.com/WordPress/gutenberg/issues/60040
2. Use the rest of the font library and check that the other install methods work as expected.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/1310626/aebc01b8-1f76-492c-94c7-8f2f94a80d0a

### CODE REVIEW

**Review by mikachan — COMMENTED** [review]
I'm not sure this fixes https://github.com/WordPress/gutenberg/issues/60040, as I'm testing using the instructions from that PR, using the Heineken Serif 18 font, and I'm still seeing the missing quotes in the preset:

image

If I add the quotes in manually the font renders correctly.

**Comment by carolinan** [comment]
I tested the PR with the same font files that I used when testing the issue, and the name is wrapped in double quotes and works in my test.

One space:
`--wp--preset--font-family--darker-grotesque: "Darker Grotesque";`

Two spaces:
`--wp--preset--font-family--ar-one-sans: "AR One Sans";`

Number and space:
`--wp--preset--font-family--slabo-13-px: "Slabo 13px";`
`--wp--preset--font-family--heineken-serif-18: "Heineken Serif 18";`


(I am using Chrome, macOS, WordPress 6.5 RC3 nightly.)

**Comment by getdave** [comment]
I'm about to test this with the fonts provided as well.

File: packages/edit-site/src/components/global-styles/font-library-modal/context.js
Comment (getdave) [inline packages/edit-site/src/components/global-styles/font-library-modal/context.js:286]:
```suggestion
					// Use font data from REST API not from client to ensure
					// correct font information is used.
					installedFontFamily.fontFace = [
```

Nit: might be worth clarifying.

**Review by getdave — APPROVED** [review]
I tested this with the provided font and another one which I had locally and it worked as described on both editor and front of site.

The font family is now quoted correctly.

Screen Shot 2024-03-25 at 11 30 20

**Comment by getdave** [comment]
@mikachan Could you possibly re-test? It seems like your result is the outlier and it would be good to know whether this is a legitimate failure 🙏

**Review by mikachan — APPROVED** [review]
Thanks for the ping! I've re-tested and this PR is fixing the issue now 🎉

Just noting that my first round of testing was done by adding a single font file rather than multiple at once. However, I've just tested both ways (with a single font and with multiple at the same time) and it's definitely working. This is good to bring in I think!

**Comment by huzaifaalmesbah** [comment]
I found that the font name is enclosed in double quotes with this PR. It is functioning properly.

Huzaifa-20240326012319

**Comment by srueegger** [comment]
I just had the time to test it too. It now works correctly for me with the PR.

**Comment by getdave** [comment]
@mcsf You raised [some potential concerns about the code of this PR in WP Slack](https://wordpress.slack.com/archives/C02QB2JS7/p1711372528859809?thread_ts=1711371204.886059&cid=C02QB2JS7). Are you able to review?

File: packages/edit-site/src/components/global-styles/font-library-modal/context.js
Comment (getdave) [inline packages/edit-site/src/components/global-styles/font-library-modal/context.js:407]: I think we could explain here _why_ we want to avoid saving to global styles post content. 

Presumably the reason this code has been added now is because we're now using the font data from the API we're getting an `id` field which wasn't present previously.

**Comment by getdave** [comment]
@bph I've marked this to be cherry picked into Gutenberg 18.0.0 in order that it will be open for _consideration_ for inclusion WP 6.5.

@matiasbenedetto @creativecoder Let's be sure to test the Gutenberg RC when it's ready to check that including this PR doesn't cause any regressions elsewhere in the _Plugin_.

File: packages/edit-site/src/components/global-styles/font-library-modal/context.js
Comment (mcsf) [reply packages/edit-site/src/components/global-styles/font-library-modal/context.js:407]: Agreed, should explain.

**Review by mcsf — APPROVED** [review]
With fresh eyes this morning, this reads as more straightforward. LGTM.

File: packages/edit-site/src/components/global-styles/font-library-modal/context.js
Comment (getdave) [reply packages/edit-site/src/components/global-styles/font-library-modal/context.js:407]: Although I think we can do that in a follow up.

**Comment by getdave** [comment]
There have been multiple reviews from contributors here and they are all ✅ 

I've gone ahead and merged this.

**Comment by getdave** [comment]
This PR was triaged by Editor release leads and contributors in WP Slack as part of the final WordPress 6.5 process.

See https://wordpress.slack.com/archives/C02QB2JS7/p1711371204886059 (requires registration).

The criteria used was:

- Was this bug identified during the RC cycle for WP 6.5?
- Is this bug directly related to one of the blessed features in this release?
- Would this bug cause existing website’s to break/fail upon upgrade to WP 6.5?
- What is the approximate scope of the impact? Scale: (Minimal) 1 - 5 (Severe).
- Is the fix introducing any new features or changes?

**Conclusion**: we propose that this fix _is_ included in WP 6.5.

**Reason**: bug affects the (blessed) Font Library feature in a severe way and this PR resolves in a clear and traceable manner.

We now await a final decision from the full release team.

**Comment by bph** [comment]
I just cherry-picked this PR to the release/18.0 branch to get it included in the next release: 5ed745a09b

**Comment by getdave** [comment]
@matiasbenedetto Please could we look into adding some test coverage for the changes in this PR. This will increase confidence of release team in including this in WP 6.5. See [related Slack discussion](https://wordpress.slack.com/archives/C065MBW03EH/p1711471479786339).

**Comment by getdave** [comment]
e2e test coverage is being worked on in https://github.com/WordPress/gutenberg/pull/60221/.

**Comment by getdave** [comment]
I just cherry picked this into WP 6.5 RC 4 branch https://github.com/WordPress/gutenberg/pull/60241/commits/d2191d4cf50d63c5f24885a8d3ab2c57a77e046c

**Pull request merged**
